AnonUpload/src/main/java/com/theironyard/controllers/AnonFileController.java

```diff
@@ -45,4 +64,19 @@ public void upload(MultipartFile file, HttpServletResponse response) throws IOEx
     public List<AnonFile> getFiles(){
         return repository.findAll();
     }
+
+    @RequestMapping(path = "/delete",method = RequestMethod.POST)
+    public void deleteFile(String fileName, String password, HttpServletResponse respone) throws Exception {
+        AnonFile fileToDelete = repository.findByFilename(fileName);
+        if(fileToDelete.getPassword().isEmpty()){
+            repository.delete(fileToDelete);
+        }else{
+            if(PasswordStorage.verifyPassword(password, fileToDelete.getPassword())){
```

Comment:
This could be reduced into 1 if statement by checking if the password exists and matches the passed in password.